packages/widgets/src/controls.tsx

```diff
@@ -0,0 +1,441 @@
+"use client";
+
+import { useCallback } from "react";
+import {
+  LoaderCircleIcon,
+  PauseIcon,
+  PlayIcon,
+  RefreshCcwIcon,
+  ServerIcon,
+  ServerOffIcon,
+  SquareIcon,
+  StepForwardIcon,
+  TriangleAlertIcon,
+} from "lucide-react";
+import { toast } from "sonner";
+import type { RouterOutput } from "@sophys-web/api-client/react";
+import { useQueue, useStatus } from "@sophys-web/api-client/hooks";
+import { api } from "@sophys-web/api-client/react";
+import { cn } from "@sophys-web/ui";
+import { Button } from "@sophys-web/ui/button";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "@sophys-web/ui/dropdown-menu";
+import { Separator } from "@sophys-web/ui/separator";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "@sophys-web/ui/tooltip";
+
+type BasicUiStatus = "online" | "offline" | "busy" | "error";
+type WorkerEnvironmentState =
+  RouterOutput["status"]["get"]["workerEnvironmentState"];
+type ManagerState = RouterOutput["status"]["get"]["managerState"];
+type CombinedEnvState = "unknown" | WorkerEnvironmentState | ManagerState;
+
+const combinedStateToBasicStatusMap: Record<CombinedEnvState, BasicUiStatus> = {
+  initializing: "busy",
+  executing_plan: "busy",
+  executing_task: "busy",
+  closing: "busy",
+  paused: "busy",
+  creating_environment: "busy",
+  starting_queue: "busy",
+  executing_queue: "busy",
+  closing_environment: "busy",
+  destroying_environment: "busy",
+  unknown: "error",
+  failed: "error",
+  idle: "online",
+  closed: "offline",
+};
+
```

Comment:
I would add something like this, to maintain consistency by capitalizing the states:
```suggestion

function capitalizeAndFormatState(state: string) {
  return state.charAt(0).toUpperCase() + state.slice(1).replaceAll("_", " ");
}

```